addon/adapters/adapter.js

```diff
@@ -156,6 +173,46 @@ export default Adapter.extend(ImportExportMixin, {
     return null;
   },
 
+  _queryFilter(record, query = {}) {
+    return keys(query).every((key) => {
+      const value = query[key],
+        matches = key.match(/(.*)Id$/),
```

Comment:
I agree with singular being more intuitive. I don't think regex solves for matching multiple cases it can do `ORs` but not `ANDs`
```
// get users with cats and dogs
query('user', filter: { pets: [{ type: 'cat' }, { type: 'dog' }] });
// get users with cats or dogs
query('user', filter: { pets: { type: /cat|dog/ } });
```